apps/laboratory/tests/shared/pages/ModalWalletPage.ts

```diff
@@ -18,20 +18,26 @@ export class ModalWalletPage extends ModalPage {
   override async switchNetwork(network: string) {
     await this.page.getByTestId('account-switch-network-button').click()
     await this.page.getByTestId(`w3m-network-switch-${network}`).click()
-    await this.page.waitForTimeout(2000)
+    const switchNetworkButton = this.page.getByTestId('account-switch-network-button')
+    await expect(switchNetworkButton).toBeVisible()
+    await expect(switchNetworkButton, `Switched network should include ${network}`).toContainText(
+      network,
+      { timeout: 10_000 }
+    )
```

Comment:
Why was this logic moved from `ModalValidator`?

Reply:
I'd change this for a regular timeout (to wait for the switch) and move the expect back to the validator

Reply:
When we use the switch network step, we should both:

- be sure it switched as expected before stepping on
- be not responsible of checking network switches in high level

I feel like making these inside the `switchNetwork` would be better in terms of encapsulation. @lukaisailovic

Reply:
That's a good point. @lukaisailovic Could you review again please?